packages/web3-eth/src/types.ts

```diff
@@ -0,0 +1,194 @@
+}
+
+interface Transaction1559UnsignedFormatted<
+	DesiredType extends ValidTypes = ValidTypes.HexString,
+	ReturnType = ValidReturnTypes[DesiredType],
+> extends BaseTransactionFormatted<DesiredType> {
+	readonly maxFeePerGas: ReturnType;
+	readonly maxPriorityFeePerGas: ReturnType;
+	readonly accessList: AccessList;
+}
+
+interface Transaction1559SignedFormatted<
+	DesiredType extends ValidTypes = ValidTypes.HexString,
+	ReturnType = ValidReturnTypes[DesiredType],
+> extends Transaction1559UnsignedFormatted<DesiredType> {
+	readonly yParity: ReturnType;
+	readonly r: ReturnType;
+	readonly s: ReturnType;
+}
+
+interface Transaction2930UnsignedFormatted<
+	DesiredType extends ValidTypes = ValidTypes.HexString,
+	ReturnType = ValidReturnTypes[DesiredType],
+> extends BaseTransactionFormatted<DesiredType> {
+	readonly gasPrice: ReturnType;
+	readonly accessList: AccessList;
+}
+
+interface Transaction2930SignedFormatted<
+	DesiredType extends ValidTypes = ValidTypes.HexString,
+	ReturnType = ValidReturnTypes[DesiredType],
+> extends Transaction2930UnsignedFormatted<DesiredType> {
+	readonly yParity: ReturnType;
+	readonly r: ReturnType;
+	readonly s: ReturnType;
+}
+
+interface TransactionLegacyUnsignedFormatted<
+	DesiredType extends ValidTypes = ValidTypes.HexString,
+	ReturnType = ValidReturnTypes[DesiredType],
+> extends BaseTransactionFormatted<DesiredType> {
+	readonly gasPrice: ReturnType;
+}
+
+interface TransactionLegacySignedFormatted<
+	DesiredType extends ValidTypes = ValidTypes.HexString,
+	ReturnType = ValidReturnTypes[DesiredType],
+> extends TransactionLegacyUnsignedFormatted<DesiredType> {
+	readonly v: ReturnType;
+	readonly r: Uint;
+	readonly s: Uint;
+}
+
+type TransactionSignedFormatted<DesiredType extends ValidTypes = ValidTypes.HexString> =
+	| Transaction1559SignedFormatted<DesiredType>
+	| Transaction2930SignedFormatted<DesiredType>
+	| TransactionLegacySignedFormatted<DesiredType>;
+
+export type TransactionInfoFormatted<
+	DesiredType extends ValidTypes = ValidTypes.HexString,
+	ReturnType = ValidReturnTypes[DesiredType],
+> = TransactionSignedFormatted<DesiredType> & {
+	readonly blockHash: HexString32Bytes | null;
+	readonly blockNumber: ReturnType | null;
+	readonly from: Address;
+	readonly hash: HexString32Bytes;
+	readonly transactionIndex: ReturnType | null;
+};
+
+export interface ReceiptInfoFormatted<
+	DesiredType extends ValidTypes = ValidTypes.HexString,
+	ReturnType = ValidReturnTypes[DesiredType],
+> {
+	readonly transactionHash: HexString32Bytes;
+	readonly transactionIndex: ReturnType;
+	readonly blockHash: HexString32Bytes;
+	readonly blockNumber: ReturnType;
+	readonly from: Address;
+	readonly to: Address;
+	readonly cumulativeGasUsed: ReturnType;
+	readonly gasUsed: ReturnType;
+	readonly contractAddress: Address | null;
+	readonly logs: Log[];
+	readonly logsBloom: HexString256Bytes;
+	readonly root: HexString32Bytes;
+	readonly status: ReturnType;
+}
+
+export interface BlockFormatted<
```

Comment:
also for block and other types defined here these are also in web3 common, we should reuse types from web3 common https://github.com/ChainSafe/web3.js/blob/2ea05a8e9b251f618bdc54f0319f0c45896822af/packages/web3-common/src/eth_execution_api.ts#L102

Reply:
Created an [issue](https://github.com/ChainSafe/web3.js/issues/4632) for this, and added it to [Web3Eth TODO list](https://github.com/ChainSafe/web3.js/issues/4608)